tests/kata-containerized-test-tool/internal/core/framework.go

```diff
@@ -0,0 +1,48 @@
+package core
+
+import (
+    "context"
+    "encoding/json"
+    "fmt"
+    "os"
+    "time"
+)
+
+type Framework struct {
+    tests     []Test
+    outputDir string
+}
+
+func NewFramework(outputDir string) *Framework {
+    return &Framework{
+        outputDir: outputDir,
+    }
+}
+
+func (f *Framework) AddTest(test Test) {
+    f.tests = append(f.tests, test)
+}
+
+func (f *Framework) RunAll(ctx context.Context) error {
+    results := make([]TestResult, 0)
+    
+    for _, test := range f.tests {
+        result := test.Run(ctx)
+        results = append(results, result)
+    }
+    
+    return f.saveResults(results)
+}
+
+func (f *Framework) saveResults(results []TestResult) error {
```

Comment:
Could you add an option to format the results in `junit` format? In the pipelines, that allows us to to [upload the results](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=740492&view=logs&j=7f72ebac-7b17-5659-0c3a-d2e7056122a8&t=f259895c-7c30-5489-6b34-e0d7bdf8f4b7) to ADO and visualize the results like [this](https://dev.azure.com/mariner-org/mariner/_TestManagement/Runs?runId=330706&_a=runCharts) . May look into https://github.com/gotestyourself/gotestsum?tab=readme-ov-file#junit-xml-output

Reply:
Agreed - is JSON ever helpful?